src/ua_parser/loaders.py

```diff
@@ -0,0 +1,120 @@
+__all__ = [
+    "load_builtins",
+    "load_data",
+    "load_yaml",
+    "MatchersData",
+    "UserAgentDict",
+    "OSDict",
+    "DeviceDict",
+]
+
+import io
+import os
+from typing import Callable, List, Optional, Tuple, Type, Union, TypedDict, Literal
+
+from .core import Matchers, UserAgentMatcher, OSMatcher, DeviceMatcher
+
+load: Optional[Callable]
+SafeLoader: Optional[Type]
+try:
+    from yaml import load, CSafeLoader as SafeLoader
+except ImportError:
+    try:
+        from yaml import load, SafeLoader
+    except ImportError:
+        load = SafeLoader = None
+
+
+class _RegexDict(TypedDict):
+    regex: str
+
+
+class UserAgentDict(_RegexDict, total=False):
+    family_replacement: str
+    v1_replacement: str
+    v2_replacement: str
+    v3_replacement: str
+    v4_replacement: str
+
+
+class OSDict(_RegexDict, total=False):
+    os_replacement: str
+    os_v1_replacement: str
+    os_v2_replacement: str
+    os_v3_replacement: str
+    os_v4_replacement: str
+
+
+class DeviceDict(_RegexDict, total=False):
+    regex_flag: Literal["i"]
+    device_replacement: str
+    brand_replacement: str
+    model_replacement: str
+
+
+MatchersData = Tuple[List[UserAgentDict], List[OSDict], List[DeviceDict]]
+
+
+def load_builtins() -> Matchers:
+    from ._matchers import MATCHERS
+
+    return MATCHERS
+
+
+def load_data(d: MatchersData) -> Matchers:
+    return (
+        [
+            UserAgentMatcher(
+                p["regex"],
+                p.get("family_replacement"),
+                p.get("v1_replacement"),
+                p.get("v2_replacement"),
+                p.get("v3_replacement"),
+                p.get("v4_replacement"),
+            )
+            for p in d[0]
+        ],
+        [
+            OSMatcher(
+                p["regex"],
+                p.get("os_replacement"),
+                p.get("os_v1_replacement"),
+                p.get("os_v2_replacement"),
+                p.get("os_v3_replacement"),
+                p.get("os_v4_replacement"),
+            )
+            for p in d[1]
+        ],
+        [
+            DeviceMatcher(
+                p["regex"],
+                p.get("regex_flag"),
+                p.get("device_replacement"),
+                p.get("brand_replacement"),
+                p.get("model_replacement"),
+            )
+            for p in d[2]
+        ],
+    )
+
+
+Yaml = Union[str, os.PathLike, io.IOBase]
+load_yaml: Optional[Callable[[Yaml], Matchers]]
+if load is None:
+    load_yaml = None
+else:
+
+    def load_yaml(path: Yaml) -> Matchers:
```

Comment:
YAML can also be used to load JSON, but maybe there should also be a json loader as it might be useful & does not need a dependency?